tests/integration/targets/win_xml/tasks/main.yml

```diff
@@ -100,6 +87,60 @@
     that:
     - not element_add_attribute_result_second is changed
 
+- name: register the sha1 of the config.xml before removing an attribute
+  ansible.windows.win_stat:
+    path: "{{ remote_tmp_dir }}\\config.xml"
+    get_checksum: yes
+  register: config_xml_checksum_before
+
+- name: try removing an attribute
+  win_xml:
+    path: "{{ remote_tmp_dir }}\\config.xml"
+    xpath: '/config/string[@key="foo"]'
+    type: attribute
+    attribute: spam
+    state: absent
+  register: attribute_remove_result
+
+- name: register the sha1 of the config.xml after removing an attribute
+  ansible.windows.win_stat:
+    path: "{{ remote_tmp_dir }}\\config.xml"
+    get_checksum: yes
+  register: config_xml_checksum_after
+
+- name: check attribute is removed
+  assert:
+    that:
+    - attribute_remove_result is changed
+    - config_xml_checksum_before.stat.checksum != config_xml_checksum_after.stat.checksum
```

Comment:
This just verifies the checksum has changed, we should be checking that the relevant attribute was removed otherwise anything could have changed.

Reply:
I would either just return the content through `slurp` or a `win_powershell` calling `Get-Content ... -Raw`. Or have a `win_powershell` inline task that validates the expected structure of the XML content, e.g. the attribute is not present.